### PR TITLE
feature autoloading bar for rest/xhr request

### PR DESCRIPTION
**Pull request by genuinefafa**
feature autoloading bar for rest/xhr request.
https://github.com/MurhafSousli/ng2-progressbar/issues/14#issuecomment-281837674

### CODE REVIEW

**Comment by MurhafSousli** [comment]
Hi @genuinefafa, I'm wondering why did you use `BrowserXhr`? in [angular API](https://angular.io/docs/ts/latest/api/http/index/BrowserXhr-class.html) it says 

> Take care not to evaluate this in non-browser contexts.

Can't we use something that works for mobile, universal and browser?

File: src/service/browser-xhr.provider.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        // TODO: do some progress magic here
+        // if (event.lengthComputable) {
+    }
+
+    // TODO: use event information to compute pending
+    // xhr.onprogress = (event) => {};
+
+    return <any>(xhr);
+  }
+
```
Comment (MurhafSousli) [inline src/service/browser-xhr.provider.ts]: this function is called before the requests are actually done, take a look

image

File: src/service/browser-xhr.provider.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  private done() {
+      this.currentRequest --;
+      if (this.currentRequest == 0)
+        this.service.done();
+  }
+}
+
```
Comment (MurhafSousli) [inline src/service/browser-xhr.provider.ts]: this function is called before the requests are actually done, take a look

image

**Comment by genuinefafa** [comment]
I'm not really sure... @MurhafSousli, I did find the idea from stackoverflow and the MDN site but it was always considering browsers, not other devices.

For the time being, browser support is better than no devices... :) Don't you think?

**Comment by MurhafSousli** [comment]
@genuinefafa I like it I will lint your PR then merge it

**Pull request merged**